engine/src/witness/sol/nonce_witnessing.rs

```diff
@@ -9,17 +9,19 @@ use crate::sol::{
 	},
 };
 use anyhow::{anyhow, Result};
+use pallet_cf_elections::vote_storage::change::MonotonicChangeVote;
 use serde_json::Value;
+use sol_prim::SlotNumber;
 use std::str::FromStr;
-
 pub async fn get_durable_nonce<SolRetryRpcClient>(
 	sol_client: &SolRetryRpcClient,
 	nonce_account: SolAddress,
-) -> Result<Option<SolHash>>
+	previous_slot: SlotNumber,
+) -> Result<Option<MonotonicChangeVote<SolHash, SlotNumber>>>
```

Comment:
nit: To keep the idea of a vote within the VoterApi, this fn should not return a vote, it can just return `(SolHash, SlotNumber)`.